students/jemielity_kamil/lesson_11_argparse_and_math/filling_the_gaps.py

```diff
@@ -0,0 +1,34 @@
+import os
+import re
+import argparse
+
+parser = argparse.ArgumentParser(description='Filling the gaps')
+parser.add_argument('-p', action='store',
+                    dest="prefix", required=True,
+                    help='prefix')
+args = parser.parse_args()
+
+
+prefix = args.prefix
+SUFFIX = '.txt'
+PATH = 'C:\\Path\\to\\folder'
+
+files_tab = []
+regex = re.compile(r'^spam\d{3}\.txt$')
+
+for _, _, files in os.walk(PATH):
+    for file in files:
+        mo = re.search(regex, file)
+        if mo is not None:
+            files_tab.append(file)
+
+files_tab = sorted(files_tab)
+number_of_files = len(files_tab)
+
+new_names = []
+for i in range(1, number_of_files + 1):
+    new_names.append(prefix + "{:03d}".format(i) + SUFFIX)
+
+for i in range(len(files_tab)):
+    if new_names[i] != files_tab[i]:
```

Comment:
Jestem przekonany, że dałoby się to zrobić wydajniej - zmieniając nazwy plików na bieżąco iterując po nich, a nie generować dodatkowej listy. Niemniej to rozwiązanie jest bardzo czytelne i zrozumiałe, więc ok! 👍